litejpeg/core/quantization.py

```diff
@@ -1,42 +1,182 @@
 # Quantization
+# In this quantization module, we are dividing our matrix with certain values as per the quantization table to increase the number of
+# zeros that is used in the compression in the next stages.
+# However instead of dividing which creates a problem because of non-storage of decimal values as such in a signal,
+# We are going to multiply the values with the inverse of the values within the matrix.
+# And give the output for the next stage.
 
 from litex.gen import *
 from litex.soc.interconnect.stream import *
+from litex.soc.interconnect.csr import *
 
 from litejpeg.core.common import *
 
+# Building up the quantization table required for the quantization module.
+# Must be specified previously.
+quant_values=[16, 11, 10, 16, 24, 40, 51, 61,
+              12, 12, 14, 19, 26, 58, 60, 55,
+              14, 13, 16, 24, 40, 57, 69, 56,
+              14, 17, 22, 29, 51, 87, 80, 62,
+              18, 22, 37, 56, 68,109,103, 77,
+              24, 35, 55, 64, 81,104,113, 92,
+              49, 64, 78, 87,103,121,120,101,
+              72, 92, 95, 98,112,100,103, 99]
 
-class Quantization(PipelinedActor, Module, AutoCSR):
-    def __init__(self, init):
-        self.sink = stream.Endpoint(block_layout(12))
-        self.source = stream.Endpoint(block_layout(12))
-        PipelinedActor.__init__(self, 3)
 
-        self._we = CSR()
-        self._addr = CSRStorage(6)
-        self._data = CSRStorage(12)
+class Quantization(PipelinedActor, Module):
+    def __init__(self):
 
-        # # #
+        # Connecting the data to the Test Bench to take the input/give the output.
+        # sink = Take the data for the module.
+        # source = Show the output specified by the module.
+        self.sink = sink = stream.Endpoint(EndpointDescription(block_layout(12)))
+        self.source = source = stream.Endpoint(EndpointDescription(block_layout(12)))
 
-        mem = Memory(12, 2**6, init=init)
-        write_port = mem.get_port(write_capable=True)
-        read_port = mem.get_port(has_re=True)
-        self.specials += mem, write_port, read_port
 
-        self.comb += [
-            write_port.adr.eq(self._addr.storage),
-            write_port.dat_w.eq(self._data.storage),
-            write_port.we.eq(self._we.re & self._we.r)
+        # Quantization ROM
+        # Storing values of the quantization tables.
+        # Since as per our discussion above we are storing the inverse of the values in the qunatization table.
```

Comment:
You don't need to say "Since a per our discussion above" - just say directly "We are storing..."